src/app/core/user/user-security/user-security.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component } from "@angular/core";
+import { DynamicComponent } from "../../view/dynamic-components/dynamic-component.decorator";
+import { OnInitDynamicComponent } from "../../view/dynamic-components/on-init-dynamic-component.interface";
+import { FormBuilder, FormControl, Validators } from "@angular/forms";
+import {
+  KeycloakAuthService,
+  Role,
+} from "../../session/auth/keycloak/keycloak-auth.service";
+import { AuthService } from "../../session/auth/auth.service";
+import { User } from "../user";
+import { MatSnackBar } from "@angular/material/snack-bar";
+import { Observable } from "rxjs";
+import { PanelConfig } from "../../entity-components/entity-details/EntityDetailsConfig";
+
+@DynamicComponent("UserSecurity")
+@Component({
+  selector: "app-user-security",
+  templateUrl: "./user-security.component.html",
+})
+export class UserSecurityComponent implements OnInitDynamicComponent {
+  form = this.fb.group({
+    username: [{ value: "", disabled: true }],
+    email: ["", [Validators.required, Validators.email]],
+    roles: new FormControl<Role[]>([]),
+  });
+  keycloak: KeycloakAuthService;
+  availableRoles: Role[] = [];
+  userId: string;
+
+  constructor(
+    private fb: FormBuilder,
+    authService: AuthService,
+    private snackbar: MatSnackBar
+  ) {
+    if (authService instanceof KeycloakAuthService) {
```

Comment:
I think the textbook, cleaner way here would be to have every AuthService having all functions (if only as dummy implementations) and maybe an "allowsEditing". So that the component here does not know or care what auth service is active.

what other AuthService implementations do we have?

Reply:
We also have the `CouchdbAuthService` with limited functionality (only edit password, no password reset, ...).

I don't see a benefit of pulling the functions up into the `AuthService` interface and then providing mock implementations in the `CouchdbAuthService`. In this case we would have to check for each function individually whether it is implemented or not (in order to adapt the UI) and just adding a `allowsEditing` flag is very vague to me, because it does not say which functions this does involve. I am not super happy with the design either but this approach at least gives us type-safety (you cannot call `createUser` on the `AuthService` unless you explicitly type-cast it).